### ISSUE TITLE
secrets patch broken in 1.10.x with userns

### ISSUE THREAD

**Issue by runcom** [projectatomic/docker#69]
after disabling selinux and running userns in docker 1.10.x I got this:
```
/ # ls -la /run/secrets/
total 0
drwxr-xr-x    2 65534    65534            6 Feb 28 16:28 .
drwxr-xr-x    3 root     root            21 Feb 28 16:28 ..
/ # touch /run/secrets/test
touch: /run/secrets/test: Permission denied
/ #
```
the `secrets` directory is owned by user `nobody` and I cannot touch anything into it (probably I shouldn't but this is another issue)
The problem seems to be the directory isn't properly chown'ed when setting up the container as it's currently happening for other directories and files mounted from the host:
```
total 28
drwx------. 4 165536 165536  149 Feb 28 11:28 .
drwx------. 8 165536 165536 4096 Feb 28 11:28 ..
-rw-r--r--. 1 root   root   2236 Feb 28 11:28 config.v2.json
-rw-r--r--. 1 root   root    903 Feb 28 11:28 hostconfig.json
-rw-r--r--. 1 165536 165536   13 Feb 28 11:28 hostname
-rw-r--r--. 1 166536 166536  174 Feb 28 11:28 hosts
-rw-r--r--. 1 165536 165536   55 Feb 28 11:28 resolv.conf
-rw-------. 1 root   root     71 Feb 28 11:28 resolv.conf.hash
drwxr-xr-x. 2 root   root      6 Feb 28 11:28 secrets
drwxrwxrwt. 2 165536 165536   40 Feb 28 11:28 shm
```
The above is the output of `ls -la` from the host into the container host directory (where we also have the secrets directory). Everything is correctly owned by 165536 who's the correct remapped user (when run with `--userns-remap=default`) except for the `secrets` directory which is owner by the host root - it should be also owned by 165536 I guess.

/cc @rhatdan

**Comment by rhatdan** [comment]
Right the problem is not SELinux but the DAC.  Directories/files owned by (real) root.

**Comment by runcom** [comment]
This issue seems because the daemon doesn't set the correct permissions on the secrets directory - while the bugzilla I filled seems to be related to selinux (I'll give you AVC asap)

**Comment by runcom** [comment]
fixed in 1.11 rebase
